packages/puppeteer-core/src/api/ElementHandle.ts

```diff
@@ -861,6 +873,35 @@ export abstract class ElementHandle<
     options?: Readonly<KeyPressOptions>
   ): Promise<void>;
 
+  async #clickableBox(): Promise<BoundingBox | null> {
+    const adoptedThis = await this.frame.isolatedRealm().adoptHandle(this);
+    const box = await adoptedThis.evaluate(element => {
+      if (!(element instanceof Element)) {
+        return null;
+      }
+      return element.getClientRects()[0]?.toJSON() as DOMRect;
```

Comment:
the previous version filtered out rects that are "too small to click". I guess we don't have a test for this and I am not sure we can add one but should we perhaps keep the filter?

Reply:
I still filter out rects using `width === 0 || height === 0`, but I realized that an element can have float values for their width and height, so I've replace this with `width < 1 || height < 1` to ensure we at least have 1 pixel.

Reply:
I don't see where the filtering happens. I see that the first `element.getClientRects()[0]` rect is taken unconditionally.

Reply:
Ah, I misunderstood what you meant. Done.